lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -80,6 +80,17 @@ LogicalResult mlir::substrait::FixedCharAttr::verify(
   return success();
 }
 
+LogicalResult mlir::substrait::FixedBinaryAttr::verify(
+    llvm::function_ref<mlir::InFlightDiagnostic()> emitError, StringAttr value,
+    FixedBinaryType type) {
+  FixedBinaryType fixedBinaryType = mlir::dyn_cast<FixedBinaryType>(type);
+  int32_t value_length = value.size();
+  if (fixedBinaryType != nullptr && value_length != fixedBinaryType.getLength())
+    return emitError() << "value length must be " << fixedBinaryType.getLength()
+                       << " characters.";
```

Comment:
Shouldn't we *fail* if `fixcedBinaryType` is a `nullptr`?
```suggestion
  if (fixedBinaryType == nullptr || value_length != fixedBinaryType.getLength())
    return emitError() << "value length must be " << fixedBinaryType.getLength()
                       << " characters.";
```
We should either (1) convince ourselves that `type` is always a `FixedBinaryType`, use a `cast` instead of a `dyn_cast` above and remove the test here or (2) use a dedicated test and error message for the `dyn_cast`.